sphinx.go

```diff
@@ -53,6 +53,20 @@ const (
 	// onion messaging jumbo onion packet.
 	MaxOnionMessagePayloadSize = 32768
 
+	// BigSizeLenMaxBytes is the maximum number of bytes required to encode
+	// a payload length using the BigSize variable-length encoding format.
+	// For TLV payloads, the length prefix can be 1, 3, 5, or 9 bytes
+	// depending on the value. Since MaxHopPayloadSize fits within a 16-bit
+	// value, 3 bytes (0xfd + 2 bytes) is the maximum needed.
+	BigSizeLenMaxBytes = 3
+
+	// MaxHopPayloadSize is the maximum size of the payload data for a
+	// single hop, excluding the BigSize length prefix and the trailing
+	// HMAC. When decoding a TLV payload, the encoded length must not
+	// exceed this value, otherwise the payload would overflow the routing
+	// info boundary into the zero-padding area used by later hops.
+	MaxHopPayloadSize = MaxRoutingPayloadSize - BigSizeLenMaxBytes - HMACSize
```

Comment:
Onion message allow for any size of payload. According to the BOLT you `SHOULD set onion_message_packet len to 1366 or 32834`, but there is nothing that requires you to do so.
In any case, the `payload` can we way bigger than 1265. It can be up to `MaxOnionMessagePayloadSize - BigSizeLenMaxBytes - HMACSize = 32703`
This also makes the check in `DecodeHopPayload` too strict.